odl/solvers/functional/default_functionals.py

```diff
@@ -2101,6 +2102,87 @@ def __repr__(self):
                                          self.__norm.outernorm.exponent,
                                          self.__norm.pwisenorm.exponent)
 
+
+class MoreauEnvelope(Functional):
+    """Moreau envelope of a convex functional.
+
+    The Moraeu evelope is a way to smooth an arbitrary convex functional
+    such that its gradient can be computed given the proximal of the original
+    functional and has the same critical points as the functional.
```

Comment:
Start new sentence: The new functional has the same critical points as the original.